src/terminal_ui.cc

```diff
@@ -1410,7 +1423,18 @@ void TerminalUI::set_ui_options(const Options& options)
     auto to_bool = [](StringView s) { return s == "yes" or s == "true"; };
 
     m_status_on_top = find("terminal_status_on_top").map(to_bool).value_or(false);
-    m_set_title = find("terminal_set_title").map(to_bool).value_or(true);
+
+    auto title_config = find("terminal_set_title").value_or("yes");
+    if (title_config != "yes" && title_config != "no" && title_config != "true" && title_config != "false")
```

Comment:
In the Kakoune codebase we use the alternate tokens for logical operator (`or`, `and` instead of `||`, `&&`)